### ISSUE TITLE
Abstract properties report inconsistent typing

### ISSUE THREAD

**Issue by blink1073** [microsoft/pyright#4364]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
This is related to #2601 but applies to a `@property` in the abstract base class.

**To Reproduce**

Create a base class that has an abstract property:

```python
import abs
from typing import Any

class Foo(abc.ABC):
    @property
    @abc.abstractmethod
    def sub_type(self) -> Any:
        pass

class Bar(Foo):
    sub_type = None
```

```
test.py:12:16 - error: Expression of type "None" cannot be assigned to declared type "property"
    Type "None" cannot be assigned to type "property" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations
```

**Expected behavior**
The property is allowed to be overridden.

**Additional context**
The code does not error when run against `mypy==0.990`.

**Comment by erictraut** [comment]
This is intended behavior. The abstract class indicates that the symbol `sub_type` must be a read-only property. In the subclass, you are attempting to implement it using a read/write attribute. The semantics for properties and attributes are similar, but they differ in important ways. If your abstract base class indicates that the type of a symbol is a property, a subclass must implement this symbol as a property for it to satisfy the base class requirements.

The following will eliminate the type violation in your code:

```python
class Bar(Foo):
    @property
    def sub_type(self):
      return None
```

**Comment by blink1073** [comment]
Makes sense, thanks @erictraut!
